src/test/java/AccountMapperTest.java

```diff
@@ -3,30 +3,131 @@
+
+       lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+       Assertions.assertEquals(expectedBattleTag, testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getBattleTag());
+    }
+
+    @Test
+    public void correctParagonLevelFetchedToDataModel() {
+
+        int expectedParagonLevel = 1111;
+
+        lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+        Assertions.assertEquals(expectedParagonLevel, testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getParagonLevel());
+    }
+
+    @Test
+    public void correctGuildNameFetchedToDataModel() {
+
+        String expectedGuildName = "Phantas Magoria";
+
+        lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+        Assertions.assertEquals(expectedGuildName, testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getGuildName());
+    }
+
+    @Test
+    public void correctHeroesListSizeFetchedToDataModel() {
+
+       heroes.add(new HeroDataModel(1, "A", "barbarian"));
+       heroes.add(new HeroDataModel(2, "B", "crusader"));
+
+       lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+       Assertions.assertEquals(2, heroes.size());
+    }
+
+    @Test
+    public void correctHighestHardcoreLevelFetchedToDataModel() {
+
+        lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+        Assertions.assertEquals(70, testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getHighestHardcoreLevel());
+    }
+
+    @Test
+    public void correctKillsFetchedToDataModel() {
+
+        mapKills.put("elites", 1974);
+
+        lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+        Assertions.assertTrue(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getKills().containsKey("elites"));
+        Assertions.assertTrue(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getKills().containsValue(1974));
+    }
+
+    @Test
+    public void wrongBattleTagFormatProvided() {
+
+       String providedBattleTag = "";
+       String wrongBattleTagFormatWarning = "Niepoprawny format battleTag! Spróbuj ponownie.";
+
+        Mockito.when(testAccountMapper.generateRequest(providedBattleTag, testFetchToken))
+                .thenReturn("Niepoprawny format battleTag! Spróbuj ponownie.");
+
+        Assertions.assertEquals(wrongBattleTagFormatWarning, testAccountMapper.generateRequest(providedBattleTag, testFetchToken));
+    }
+
+    @Test
+    public void providedBattleTagDoesntExistThrowsException() {
+
+        AccountMapper testAccountMapper = new AccountMapper();
```

Comment:
Why the new instances here?

Reply:
If I am using @Mock instances here i get errors:
AssertionFailedError: Expected java.lang.NullPointerException to be thrown, but nothing was thrown.
So I use separately regular instances of those classes. I moved them at the begginig of test class where variables are declared.